### PR TITLE
Early CANopen protocol support

### PR DESCRIPTION
**Pull request by p-l-**

### CODE REVIEW

**Comment by p-l-** [comment]
@guedou maybe you can review this PR since it does not touch `arch/` and one OSX test has passed, and we can stop the tests on this one?

**Comment by guedou** [comment]
@smainand is currently trying it with his own packets =)

**Comment by smainand** [comment]
Hello,

thanks @p-l- it works great with the wireshark example. I think you can safely merge the "unknown" and "identifier" fields in one ``` XBitField("identifier", 0, 29) ``` ; the socketCAN format seems to use this to handle both CAN2.0A (11 bits) and B (29 bits) identifiers in a same sized packet.

In my case I'm playing with can-utils, and I'm still trying to understand why the identifier bytes are swapped, and how wireshark deals with it ([iso-tp_obd_can2.zip](https://github.com/secdev/scapy/files/1646528/iso-tp_obd_can2.zip)). Packets get wrapped in a Linux cooked layer with this setup so maybe you could add a ``` bind_layers(CookedLinux, CAN, proto=12) ``` so that the CAN dissector get called right away.

Have you thought of a way to handle in Scapy different CAN based protocols, such as iso15765-2, UDS, CANopen ? Maybe with a conf.contrib['can'].proto key ?

Seb.

File: scapy/layers/can.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
+# This file is part of Scapy
+# See http://www.secdev.org/projects/scapy for more informations
+# Copyright (C) Philippe Biondi <phil@secdev.org>
```
Comment (guedou) [inline scapy/layers/can.py]: I think that you should write your name here =)

Comment (p-l-) [reply scapy/layers/can.py]: I'm giving the (c) to @phil777 :)

**Comment by gpotter2** [comment]
How about adding "can" to `config.conf.load_layers` ?

**Comment by p-l-** [comment]
@gpotter2 I prefer to keep the list of automatically loaded protocols small, and I think when people want to dissect CAN, they'll know it. Also, for now, the CAN layer is not bound to anything that could arrive from the network (Ether, IP, UDP, TCP, etc.). What do you think?

File: scapy/layers/can.py (hunk 1 above)
Comment (guedou) [reply scapy/layers/can.py]: Too kind, or too lazy ? =)

**Pull request merged**